### DISCUSSION TITLE
Is there a way to bulk connect rear ports between patch panels?

### DISCUSSION THREAD

**Discussion by lonelyadmin** [netbox-community/netbox#8493] [Q&A]
After following [this awesome post](https://cs7networks.co.uk/2019/09/28/netbox-creating-patch-panels-and-passive-cabling/) I got to steps 7-9 and realizing that there's going to be quite a bit of clicking.

Is there a way to bulk connect rear ports between patch panels?

**Comment by DanSheps** [comment]
A netbox script would do it, will require some python coding however.

**Reply by lonelyadmin** [reply]
Are there any examples connecting rear-ports?

**Reply by candlerb** [reply]
Already done :-)

https://github.com/netbox-community/reports/blob/master/scripts/multi_connect.py

Just drop this file in `/opt/netbox/netbox/scripts/`

**Reply by tlourey** [reply]
Is there any documentation for this script?

**Reply by tlourey** [reply]
> Is there any documentation for this script?

Disregard. i've never run a netbox script before. I retract the question.

**Comment by jeremystretch** [comment]
The simplest way is to create a spreadsheet with all the desired cables and upload it to NetBox as CSV data.

**Reply by lonelyadmin** [reply]
Trying this approach as well. I exported what I have, but this import columns don't really match up, specifically for the termination points. I don't see where to define that in the import csv.

**Reply by candlerb** [reply]
It's a bit tricky to get right, since as you observe the CSV import and export formats are not compatible with each other.

I wrote an export template [here](https://github.com/netbox-community/netbox/discussions/7792#discussioncomment-1631444) to export existing cables in a CSV form which is compatible with import.  I use this when I need to bulk-change one end of multiple cables (delete them and re-import them).  Although I've only tested with interface-to-interface and interface-to-frontport, it ought to work with rearport-to-rearport.

Using that template as a guide, I think an example of a valid cable between device pp1 rearport R1 and device pp2 rearport R1 is:

```
side_a_device,side_a_type,side_a_name,side_b_device,side_b_type,side_b_name,type,status,label,color,length,length_unit
pp1,dcim.rearport,R1,pp2,dcim.rearport,R1,cat5e,connected,"",111111,500.00,cm
```

**Reply by jeremystretch** [reply]
The cables import view, like all CSV import forms, includes a detailed description of all supported columns and acceptable values.

**Comment by mbirth** [comment]
You can create 1 rear port with 24 "positions", map the positions to the front ports and then just connect this one rear "trunk" port to the target panel's rear port.
